hazelcast/src/main/java/com/hazelcast/instance/HazelcastProperties.java

```diff
@@ -55,34 +56,46 @@
      * @param hazelcastProperties array of {@link HazelcastProperty} to configure
      */
     protected void initProperties(Properties properties, HazelcastProperty[] hazelcastProperties) {
+        this.properties = properties;
         for (HazelcastProperty property : hazelcastProperties) {
```

Comment:
Hmm, I know you haven't changed this, but Properties is mutable. Don't we want' to guard it?